specs/order_spec.rb

```diff
@@ -79,31 +79,78 @@
 end
 
 # TODO: change 'xdescribe' to 'describe' to run these tests
-xdescribe "Order Wave 2" do
+describe "Order Wave 2" do
   describe "Order.all" do
     it "Returns an array of all orders" do
-      # TODO: Your test code here!
-      # Useful checks might include:
-      #   - Order.all returns an array
-      #   - Everything in the array is an Order
-      #   - The number of orders is correct
-      #   - The ID and products of the first and last
-      #       orders match what's in the CSV file
-      # Feel free to split this into multiple tests if needed
-    end
+		orders = Grocery::Order.all
+		orders.must_be_kind_of Array
+	end
+	
+	it "Confirms everything in the array is of type Order" do 
+		orders = Grocery::Order.all
+		orders.each { |i|
+			i.must_be_kind_of Grocery::Order
+		}
+	end
+	
+	it "Confirms the number of orders is correct"  do 
+		orders = Grocery::Order.all
+		rawCount = 0
+		CSV.foreach('../support/orders.csv') { |row|
+			rawCount += 1
+		}
+		
+		orderCount = 0
+		orders.each { |i|
+			orderCount += 1
+		}
+		
+		rawCount.must_equal orderCount
+	end
   end
 
   describe "Order.find" do
     it "Can find the first order from the CSV" do
-      # TODO: Your test code here!
+		myOrders = []
+      CSV.foreach('../support/orders.csv') { |row|
+				order_prodcuts = {}
+				order_id = row[0] #this is the ID
```

Comment:
I'm curious why you included this big loop to read all the orders here. You put them all into the `myOrders` array, but as far as I can tell it isn't ever used.

The last two lines of this test seem to do everything you need (trying to find the first one and checking that the ID is as expected).